src/katgpucbf/xbgpu/beamform.py

```diff
@@ -147,22 +169,27 @@ def __init__(
         weights_dims = (n_ants, accel.Dimension(n_beams, exact=True))
         self.slots["weights"] = accel.IOSlot(weights_dims, np.complex64)
         self.slots["delays"] = accel.IOSlot(weights_dims, np.float32)
-        self.slots["rand_states"] = accel.IOSlot(
-            (
-                accel.Dimension(n_batches, exact=True),
-                accel.Dimension(n_channels_per_substream, exact=True),
-                accel.Dimension(n_spectra_per_batch, exact=True),
-            ),
-            RAND_STATE_DTYPE,
-        )
-        rand_states = builder.make_states(
-            command_queue,
-            (n_batches, n_channels_per_substream, n_spectra_per_batch),
-            seed=seed,
-            sequence_first=sequence_first,
-            sequence_step=sequence_step,
-        )
-        self.bind(rand_states=rand_states)
+        if self._dither_enabled:
+            self.slots["rand_states"] = accel.IOSlot(
+                (
+                    accel.Dimension(n_batches, exact=True),
+                    accel.Dimension(n_channels_per_substream, exact=True),
+                    accel.Dimension(n_spectra_per_batch, exact=True),
+                ),
+                RAND_STATE_DTYPE,
+            )
+            rand_states = builder.make_states(
+                command_queue,
+                (n_batches, n_channels_per_substream, n_spectra_per_batch),
+                seed=seed,
+                sequence_first=sequence_first,
+                sequence_step=sequence_step,
+            )
+            self.bind(rand_states=rand_states)
+
+    @property
+    def _dither_enabled(self) -> bool:
+        return any(beam.dither != DitherType.NONE for beam in self.template.beams)
```

Comment:
I know it absolutely makes sense for this to be here (in `Beamform`), but
* considering the exact same incantation is in `beamform.mako`,
* would you be opposed to making this a property of `BeamformTemplate`?
  * and passing the `any_dither` bool into `accel.build`.
* It would also take the arglist back down to two parameters (if that matters).
  * You wouldn't need to pass `DitherType` anymore.